datadog_checks_base/datadog_checks/checks/prometheus/mixins.py

```diff
@@ -345,134 +416,127 @@ def _extract_metric_from_map(self, _m, messages, obj_map, obj_help):
+                    elif label.name in scraper_config['label_joins'][message.name]['labels_to_get']:
                         labels_list.append((label.name, label.value))
                 try:
-                    self._label_mapping[matching_label][matching_value] = labels_list
+                    scraper_config['_label_mapping'][matching_label][matching_value] = labels_list
                 except KeyError:
                     if matching_value is not None:
-                        self._label_mapping[matching_label] = {matching_value: labels_list}
+                        scraper_config['_label_mapping'][matching_label] = {matching_value: labels_list}
 
-    def join_labels(self, message):
+    def _join_labels(self, message, scraper_config):
         # Filter metric to see if we can enrich with joined labels
-        if self.label_joins:
+        if scraper_config['label_joins']:
             for metric in message.metric:
                 for label in metric.label:
-                    if label.name in self._watched_labels:
+                    if label.name in scraper_config['_watched_labels']:
                         # Set this label value as active
-                        if label.name not in self._active_label_mapping:
-                            self._active_label_mapping[label.name] = {}
-                        self._active_label_mapping[label.name][label.value] = True
+                        if label.name not in scraper_config['_active_label_mapping']:
+                            scraper_config['_active_label_mapping'][label.name] = {}
+                        scraper_config['_active_label_mapping'][label.name][label.value] = True
                         # If mapping found add corresponding labels
                         try:
-                            for label_tuple in self._label_mapping[label.name][label.value]:
+                            for label_tuple in scraper_config['_label_mapping'][label.name][label.value]:
                                 extra_label = metric.label.add()
                                 extra_label.name, extra_label.value = label_tuple
                         except KeyError:
                             pass
 
-    def process_metric(self, message, **kwargs):
+    def process_metric(self, message, scraper_config, metric_transformers=None):
         """
         Handle a prometheus metric message according to the following flow:
-            - search self.metrics_mapper for a prometheus.metric <--> datadog.metric mapping
+            - search scraper_config['metrics_mapper'] for a prometheus.metric <--> datadog.metric mapping
             - call check method with the same name as the metric
             - log some info if none of the above worked
 
-        `send_histograms_buckets` is used to specify if yes or no you want to send the buckets as tagged values when dealing with histograms.
+        `metric_transformers` fix me
```

Comment:
Thanks! I was discussing this with @devonboyer whether or not we should include `metric_transformers` as an argument to the `process` function or as an element in the `scraper_config`.

We decided that it would be better to pass it in as an argument so it is similar to how it was done before.